### ISSUE TITLE
iOS release memory leak

### ISSUE THREAD

**Issue by a-koka** [zmxv/react-native-sound#503]
The `release` method for iOS needs to change as below:

` [[self callbackPool] removeObjectForKey:player];`

to 

` [[self callbackPool] removeObjectForKey:key];`

**Comment by Winglonelion** [comment]
Hi @a-koka, seem like this project no longer maintained. I planning to make a new project to completly manage and play sound with some new features base on this project and our fork. But i don't have a deep knowgle in `objective-C`.
So, could you describe your issue and your solution more clarify? 

Hope to receive a helping hand from you.

**Comment by a-koka** [comment]
Sure, I'd be happy to give a helping hand. I have a fork myself. Curious what features/fixes your fork introduces?

Regarding my proposed change, it's a simple typo fix in `RNSound.m`. Look at the function below where `removeObjectForKey` is using an incorrect `key`. That's all.

```
RCT_EXPORT_METHOD(release:(nonnull NSNumber*)key) {
  AVAudioPlayer* player = [self playerForKey:key];
  if (player) {
    [player stop];
    [[self callbackPool] removeObjectForKey:player];
    [[self playerPool] removeObjectForKey:key];
    NSNotificationCenter *notificationCenter = [NSNotificationCenter defaultCenter];
    [notificationCenter removeObserver:self];
  }
}
```

**Comment by Winglonelion** [comment]
you can check my fork here: https://github.com/nodejsaber/react-native-sound

i merged some fixs in pull request and rewrote js base on es6 syntax. Add Eslint to lint project.

Convert play pool key to double and unique it to prevent Media life circle lock when play a sound multi times.

So, i planning to convert all api to Promise/ Async - Await.

Add some api follow Factory pattern. You have a Factory class and diffrent type of resource need diffrent api to create sound instant. (It exactly what i'm done in our Project.)

i think it will make this library ease to use and ease to know than current. 

This thinking help to manage sounds from other resources ease, and separate some specify handles of diffrent resources.

Add timeout option when create new sound from online resouces to prevent case server not respond -> resolve Promise lock when create new sound from online resources.

Update late ...

**Comment by slycoder** [comment]
For what it's worth,  I opened #476 a while ago but no one has merged it yet =(.

**Comment by paulmelnikow** [comment]
Looks like this was fixed in #476.
